### PR TITLE
REF: rename 'labels' to 'codes' in algorithms.safe_sort and ._factorize

### PR DESCRIPTION
**Pull request by topper-123**
Finishes up the renaming started by #29509 in ``core.algorithms`` of 'labels' to 'codes.' Also renames in  related tests.

### CODE REVIEW

File: pandas/core/algorithms.py [hunk 1]
```diff
@@ -468,18 +470,16 @@ def _factorize_array(values, na_sentinel: int = -1, size_hint=None, na_value=Non
 
     Returns
     -------
-    labels : ndarray
+    codes : ndarray
```
Comment (jbrockmendel) [inline pandas/core/algorithms.py]: do we necessarily know the dtype?  e.g. could this be `ndarray[int64]`?

Comment (topper-123) [reply pandas/core/algorithms.py]: As it's implemented, codes will always be of type ``ndarray[int64]``. I've got a suspicion taht there are cases where would like it to return arrays of fewer bits, but yes, as it is now, it always is an int64 array.

File: pandas/core/algorithms.py [hunk 2]
```diff
@@ -1924,51 +1924,52 @@ def diff(arr, n: int, axis: int = 0):
 #  this module.
 def safe_sort(
     values,
-    labels=None,
+    codes=None,
```
Comment (jreback) [inline pandas/core/algorithms.py]: followup to type this would be great

**Comment by jreback** [comment]
thanks @topper-123 if you'd like to followup with typing where indicated would be great.

**Pull request merged**